src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -2447,6 +2447,10 @@ private void seekPastOversizedValue() {
 
     @Override
     public Event nextValue() {
+        if (event != Event.NEEDS_DATA) {
+            annotationSequenceMarker.startIndex = -1;
+            annotationSequenceMarker.endIndex = -1;
+        }
```

Comment:
🗺️ As long as we're calling `nextValue()` after having a non-`NEEDS_DATA` event, we should be resetting this to ensure that it does not leak into other values. Before this change, the cursor was leaking state between values for annotations when there are 3 or more annotations followed by a value with 2 or fewer annotations.
 - Occurred for both SID and FlexSym annotations. I.e.:
   - `E0 01 01 EA E9 0D FF 61 FF 62 FF 63 60 E7 FF 64 60`
   - `E0 01 01 EA E4 07 D8 0F B6 91 61 91 62 91 63 E6 07 15 17 19 60 E4 15 60`
   - ^should both be `$ion_1_1 a::b::c::0 d::0`, but were being read as `$ion_1_1 a::b::c::0 a::b::c::0`
 - Also occurs when there is an unannotated value in the middle. For example `$ion_1_1 a::b::c::0 false d::0` was being read as `$ion_1_1 a::b::c::0 false a::b::c::0`